app/controllers/orders_controller.rb

```diff
@@ -1,11 +1,32 @@
 class OrdersController < ApplicationController
-  def create
+  def new
     @order = Order.new
+    @order.build_shipping_address
+    @order.build_billing_address
+  end
+
+  def create
+    @order = Order.new(order_params)
     @order.add_items_from(cart)
+    @order.user = current_user
     if @order.save
       cart.empty
       session[:checkout_order_id] = @order.id
       redirect_to new_charge_path
     end
   end
+
+  def show
+    @order = Order.find_by(id: params[:id])
+    @order_items = @order.order_items
```

Comment:
I'd have to look at the logs to see if this is necessary, but an `includes` on line 20 might save a hit on the database here.